ultravox/inference/infer.py

```diff
@@ -30,6 +32,60 @@ def __init__(
         self.processor = processor
         self.dtype = dtype
 
+    def batch_infer(
+        self,
+        all_samples: List[datasets.VoiceSample],
+        batch_size: int = 1,
+        max_tokens: Optional[int] = None,
+        temperature: Optional[float] = None,
+    ) -> base.InferenceGenerator:
```

Comment:
If `all_samples` is a list of `datasets.VoiceSample`, it means all audio files need to be loaded in memory. It will not scale for large datasets.

Reply:
Ah yes, i forgot about this, i need to make sure not to use lists anywhere when dealing with the datasets. Its tricky because I originally used the .map funciton but that seemed to corrupt the data it was mapping.